sentry-ruby/lib/sentry/client.rb

```diff
@@ -80,8 +80,9 @@ def capture_event(event, scope, hint = {})
     # Initializes an Event object with the given exception. Returns `nil` if the exception's class is excluded from reporting.
     # @param exception [Exception] the exception to be reported.
     # @param hint [Hash] the hint data that'll be passed to `before_send` callback and the scope's event processors.
+    # @param mechanism [Mechanism] the mechanism that holds a type and whether the exception was handled or not.
     # @return [Event, nil]
-    def event_from_exception(exception, hint = {})
+    def event_from_exception(exception, hint = {}, mechanism = nil)
```

Comment:
Can we make `mechanism` part of the `hint` here? 
The current approach limits the ways to pass hint, from both `key: value` and `{ key: value }`, to just the latter. IMO it's a breaking change (albeit a minor one), and it's not really necessary. If we simply treat `mechanism` the same as `ignore_exclusions` here, then we won't have the problem.